dcos_installer/backend.py

```diff
@@ -62,10 +58,13 @@ def generate_node_upgrade_script(installed_cluster_version, config_path=CONFIG_P
         return 1
 
     config = Config(config_path)
-    validation = validate_gen(config)
-    if not validation:
+    try:
```

Comment:
This stanza is identical to the one in `do_configure()`, and these functions are intended to do the same thing, except that `generate_node_upgrade_script()` outputs an additional artifact. So I think it'd be worth it to factor this bit of code into a helper function and call it in both places.

Reply:
If you don't mind I'd like to keep it as it is. I don't see any good and useful abstraction I could build out of this code block.